fem/bilinearform_ext.cpp

```diff
@@ -115,7 +115,11 @@ void PABilinearFormExtension::AssembleDiagonal(Vector &y) const
    Array<BilinearFormIntegrator*> &integrators = *a->GetDBFI();
 
    const int iSz = integrators.Size();
-   if (elem_restrict)
+   bool use_restrict = elem_restrict;
+#ifdef MFEM_USE_CEED
+   use_restrict = use_restrict && !DeviceCanUseCeed();
+#endif
```

Comment:
I don't think you need to guard this, is it creating issues?

Reply:
I can check. I wasn't sure if DeviceCanUseCeed was only in the Ceed part of the code.

Reply:
This condition is already used in `Mult`, factorizing this condition into a private function would be cleaner I think.

Reply:
You are correct. I dropped the guard.